Tasks/ExtractFiles/extractFiles.ts

```diff
@@ -0,0 +1,307 @@
+    }
+
+    // search backwards from the first wild card char for the nearest path separator
+    for (var i = firstWildIndex - 1; i > -1; i--) {
+        if (normalizedPattern.charAt(i) == path.sep) {
+            return {
+                file: null,
+                directory: makeAbsolute(normalizedPattern.substring(0, i + 1)),
+                search: normalizedPattern.substring(i + 1, normalizedPattern.length)
+            };
+        }
+    }
+
+    console.log('No path specified for search pattern: ' + normalizedPattern + ' defaulting to: ' + repoRoot);
+
+    return {
+        file: null,
+        directory: repoRoot,
+        search: normalizedPattern
+    };
+}
+
+function makeAbsolute(normalizedPath: string): string {
+    tl.debug('makeAbsolute:' + normalizedPath);
+
+    var result = normalizedPath;
+    if (!path.isAbsolute(normalizedPath)) {
+        result = repoRoot + path.sep + normalizedPath;
+        console.log('Relative file path: '+ normalizedPath+' resolving to: ' + result);
+    }
+    return result;
+}
+
+// Find matching archive files
+var files: string [] = findFiles();
+console.log('Found: ' + files.length + ' files to extract:');
+for (var i = 0; i < files.length; i++) {
+    console.log(files[i]);
+}
+
+// Clean the destination folder before extraction?
+if (cleanDestinationFolder && tl.exist(destinationFolder)) {
+    console.log('Cleaning destination folder before extraction: ' + destinationFolder);
+    tl.rmRF(destinationFolder, false);
+}
+    
+// Create the destination folder if it doesn't exist
+if (!tl.exist(destinationFolder)) {
+    console.log('Creating destination folder: ' + destinationFolder);
+    tl.mkdirP(destinationFolder);
+}
+
+function isZip(file) {
+    return file.endsWith('.zip')
+        || file.endsWith('.jar')
+        || file.endsWith('.war')
+        || file.endsWith('.ear');
+}
+
+function isTar(file) {
+    var name = win ? file.toLowerCase() : file;
+    // standard gnu-tar extension formats with recognized auto compression formats
+    // https://www.gnu.org/software/tar/manual/html_section/tar_69.html
+    return name.endsWith('.tar')      // no compression
+        || name.endsWith('.tar.gz')   // gzip
+        || name.endsWith('.tgz')      // gzip
+        || name.endsWith('.taz')      // gzip
+        || name.endsWith('.tar.Z')    // compress
+        || (win && name.endsWith('tar.z')) // no case comparison for win
+        || name.endsWith('.taZ')      // compress // no case for win already handled above
+        || name.endsWith('.tar.bz2')  // bzip2
+        || name.endsWith('.tz2')      // bzip2
+        || name.endsWith('.tbz2')     // bzip2
+        || name.endsWith('.tbz')      // bzip2
+        || name.endsWith('.tar.lz')   // lzip
+        || name.endsWith('.tar.lzma') // lzma
+        || name.endsWith('.tlz')      // lzma
+        || name.endsWith('.tar.lzo')  // lzop
+        || name.endsWith('.tar.xz')   // xz
+        || name.endsWith('.txz');     // xz
+}
```

Comment:
Can we also allow other file types supported by 7-Zip, such as .rar?  Here is the complete list from their website:

Packing / unpacking: 7z, XZ, BZIP2, GZIP, TAR, ZIP and WIM
Unpacking only: AR, ARJ, CAB, CHM, CPIO, CramFS, DMG, EXT, FAT, GPT, HFS, IHEX, ISO, LZH, LZMA, MBR, MSI, NSIS, NTFS, QCOW2, RAR, RPM, SquashFS, UDF, UEFI, VDI, VHD, VMDK, WIM, XAR and Z.